NMS-14844: Documentation for Cloud Services Connector and Time Series DB (#5462)
OpenNMS/opennms: docs/modules/deployment/pages/time-series-storage/timeseries/hosted-tss.adoc

= Time Series Service

> The time series service provides a scalable cloud-hosted time series database to store and persist performance metrics that {page-component-title} collects from the devices it monitors.
> 
> In addition, users can view time series summary data including health status, capacity, and metrics per second in the OpenNMS Portal.

Reviewer: I would suggest combining these paragraphs.

```suggestion
The time series service provides a scalable cloud-hosted time series database to store and persist performance metrics that {page-component-title} collects from the devices it monitors.
In addition, users can view time series summary data including health status, capacity, and metrics per second in the OpenNMS Portal.
```

== Requirements

You must have the following to use the time series service:

* Horizon 31+ (Meridian availability with the 2023 release)
* Time series service subscription that provides access to the https://portal.opennms.com[OpenNMS Portal] to retrieve your activation key and view time series service metrics

Contact sales@opennms.com to sign up for the time series service.

== Deployment

There are three phases of deployment to set up the time series service for your system:

> * Update OpenNMS Cloud Plugin files
> * Obtain an activation key for the service
> * Enable cloud services connectivity

Reviewer: ```suggestion
* Update OpenNMS Cloud Plugin files.
* Obtain an activation key for the service.
* Enable cloud services connectivity.
```

=== Update OpenNMS Cloud Plugin files

The OpenNMS Cloud Plugin makes it possible for a {page-component-title} instance to send time series data to an OpenNMS-hosted cloud service.

> This plugin is installed by default in {page-component-title} 31.

Reviewer: ```suggestion
This plugin is installed by default in Horizon 31.
```

> However, you need to move the configuration files into the correct location and restart {page-component-title} as part of the time series service set up:

Reviewer: ```suggestion
However, you need to move the configuration files into the correct location and restart {page-component-title} as part of the time series service setup process:
```

[source, console]
----
cp /opt/opennms/etc/examples/opennms.properties.d/timeseries.properties /opt/opennms/etc/opennms.properties.d/timeseries.properties

systemctl restart openmns
----

> === Obtain an activation key for time series services

Reviewer: ```suggestion
=== Obtain an activation key
```

The activation key lets {page-component-title} connect with OpenNMS cloud services (currently the time series service).

You must log in to the OpenNMS Portal to get this activation key, copy it, and then paste it into the appropriate screen in {page-component-title}.

When you signed up for the time series service, you received an activation email from OpenNMS with a login for the OpenNMS Portal.

To obtain the activation key, follow these steps:

. If you have not already done so, activate your OpenNMS Portal account by clicking the *Account Activation* button in your welcome email.

> . On the Create Your OpenNMS Account screen, type a password in the *Password* box.
> . Click *Create My Account*.

Reviewer: ```suggestion
. On the Create Your OpenNMS Account screen, type a password in the *Password* box and click *Create My Account*.
```

. Sign in to the OpenNMS Portal with your username and password.
. Choose *OpenNMS Core Instances* in the left menu and click *Add New*.
. Type in the name of your instance and click *Generate Activation Key*.
. Copy the instance key and click *Save & Close*.

+
NOTE: The key is valid for one hour.

> If you do not use it within that time, you must log back into the OpenNMS Portal and copy it from your instance.

Reviewer 2: Users cannot renew/generate another one-time code (such functionality does not exist).

Workaround: they'll have to delete the old instance and create another.

> . Copy the instance key and click *Save & Close*.
> 
> +
> NOTE: The key is valid for one hour.
> If you do not use it within that time, you must log back into the OpenNMS Portal and copy it from your instance.

Reviewer: ```suggestion
. Copy the instance key and click *Save & Close*.
+
NOTE: The key is valid for one hour.
If you do not use it within that time, you must log back into the OpenNMS Portal and copy it from your instance.
```

. Log in to the OpenNMS {page-component-title} web UI for your instance.

> . In the main menu, click *Info>Cloud Services*.

Reviewer 2: Based on the wording here, it looks like CLOUD-2681 is taken into account.  Can you please confirm @Bonrob2 ?

Author: Yes, the docs take that ticket into account.

> . In the main menu, click *Info>Cloud Services*.

Reviewer: ```suggestion
. Click menu:Info[Cloud Services] in the top menu bar.
```

For more information on how to view this information, see <<view-time-series-data,View Time Series Summary Data>>.
To change the data collection frequency, see xref:operation:deep-dive/performance-data-collection/collectd/collection-packages.adoc#ga-collectd-packages-services[Service configuration attributes].

== Troubleshooting

To verify that your time series service is set up correctly, run the `health-check` command in Karaf.
You should see something like the following:

[source,karaf]
----
admin@opennms()> health-check
Verifying the health of the container

Verifying installed bundles                    [ Success  ]
Cloud status                                   [ Success  ] => Cloud status=SERVING
Connecting to ElasticSearch ReST API (Flows)   [ Success  ] => Not configured

=> Everything is awesome
----

Check Karaf logs if the registration fails or the health check does not say `SERVING`.
If this is the case, you will need to open a support ticket.

You can also check that `OPENNMS_HOME/share/rrd` files are still being updated.
If they are not, this indicates that the time series service is not working.

[[view-time-series-data]]

> == View time series summary data

Reviewer: ```suggestion
== View time series summary data

```
